pkg/cloud/services/eks/nodegroup.go

```diff
@@ -246,6 +248,10 @@ func (s *NodegroupService) createNodegroup() (*awseks.Nodegroup, error) {
 		return nil, errors.New("unable to create nodegroup without launch template")
 	}
 
+	if managedPool.LaunchTemplate != nil && managedPool.LaunchTemplate.ID == "" {
```

Comment:
can probably just be an `|| managedPool.LaunchTemplate.ID == ""` added to the above check.